src/redux/slices/user.js

```diff
@@ -371,6 +372,50 @@ export const getICNSData = createAsyncThunk(
   }
 );
 
+export const addConnectedApp = createAsyncThunk(
+  'keyring/addConnectedApp',
+  /**  @param {any} [app] */
+  async (app, { getState, dispatch }) => {
+    const currentConnectedApps = getState().user.connectedApps;
+    const { name, canisterList, lastConection } = app;
+    let connectedApps = [app, ...currentConnectedApps];
+
+    const appAlreadyAdded = currentConnectedApps.find(
+      connectedApp => connectedApp.name === name
+    );
+
+    if (appAlreadyAdded) {
+      connectedApps = currentConnectedApps.map(connectedApp =>
+        connectedApp.name === name
+          ? {
+              lastConection,
+              canisterList: [...canisterList, ...connectedApp.canisterList],
+              ...connectedApp,
+            }
+          : connectedApp
+      );
+    }
+
+    dispatch(setConnectedApps(connectedApps));
```

Comment:
why not returning the list and use `addConnectedApp.fulfilled` to set the list? Instead of a new dispatch